OPHD/UI/IconGrid.cpp

```diff
@@ -378,10 +378,40 @@ void IconGrid::update()
 		if (mShowTooltip)
 		{
 			const auto& highlightedName = mIconItemList[mHighlightIndex].name;
-			const auto tooltipRect = NAS2D::Rectangle<int>{{position.x, position.y - 15}, {mFont.width(highlightedName) + 4, mFont.height()}};
-			renderer.drawBoxFilled(tooltipRect, NAS2D::Color{245, 245, 245});
+
+			int toolTipBoxWidth = mFont.width(highlightedName);
+			int toolTipBoxHeight = mFont.height();
+
+			auto resourceCostIterator = mResourceCosts.find(mIconItemList[mHighlightIndex].name);
+			if (resourceCostIterator != mResourceCosts.end())
+			{
+				for(auto& resourceCost: resourceCostIterator->second)
+				{
+					toolTipBoxWidth = std::max(toolTipBoxWidth, mFont.width(std::get<0>(resourceCost) + ": " + std::to_string(std::get<1>(resourceCost))));
+				}
```

Comment:
You can potentially use [structured bindings](https://en.cppreference.com/w/cpp/language/structured_binding) to unpack the tuple in the ranged-for, without having to use `std::get<index>`.

```cpp
for(const auto& [resourceName, resourceCost] : resourceCostIterator->second)
{
	toolTipBoxWidth = std::max(toolTipBoxWidth, mFont.width(resourceName + ": " + std::to_string(resourceCost)));
}
```